libusb/os/darwin_usb.c

```diff
@@ -2332,27 +2306,36 @@ static int darwin_handle_transfer_completion (struct usbi_transfer *itransfer) {
   return usbi_handle_transfer_completion (itransfer, darwin_transfer_status (itransfer, tpriv->result));
 }
 
-#if !defined(HAVE_CLOCK_GETTIME)
 void usbi_get_monotonic_time(struct timespec *tp) {
-  mach_timespec_t sys_time;
+// If the SDK is new enough to have the clock_gettime() declaration, and the deplopment target is at least 10.12...
+#if ((MAC_OS_X_VERSION_MAX_ALLOWED >= 101200) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 101200))
```

Comment:
Isn't MAC_OS_X_VERSION_MAX_ALLOWED always >=  MAC_OS_X_VERSION_MIN_REQUIRED? So that only the latter needs checking?

Reply:
Hmm, it generally would be, but I'm not 100% sure it *must* be, so I'd argue to keep it like this.

Reply:
OK. Just another nitpick: `/* comments */` are standard except inside Windows files.

Reply:
"deplopment" :) And the ellipses are unneeded. Maybe rephrase to something shorter and less redundant, e.g. "If the SDK declares clock_gettime(), and the deployment target supports it"

From what I understand MAC_OS_X_VERSION_MAX_ALLOWED is the highest version that the SDK knows about, and the target cannot be set higher. But I am fine with keeping both checks because I don't know for sure. LOL, I googled that macro now and this came on top :) https://lists.apple.com/archives/xcode-users/2005/Aug/msg00399.html